src/charm.py

```diff
@@ -147,10 +155,24 @@ def _on_munge_available(self, event):
             self.unit.status = BlockedStatus("Error restarting munge")
             event.defer()
 
-    def _on_db_unavailable(self, event):
-        self._stored.db_info = {}
-        # TODO tell slurmctld that slurmdbd left?
-        self._check_status()
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Set database info in StoredState and invoke _write_config_and_restart_slurmdbd.
+
+        Args:
+            event (DatabaseCreatedEvent):
+                Information passed by MySQL after the slurm_acct_db database has been created.
+        """
+        logger.debug("Configuring new backend database for slurmdbd.")
+        self.set_db_info(
+            {
+                "db_username": event.username,
+                "db_password": event.password,
+                "db_hostname": "127.0.0.1",
```

Comment:
Why do you use `localhost` as `db_hostname`? Shouldn't it be the IP of the node where MySQL is running?

Reply:
The mysql-router runs a local proxy that listens on 127.0.0.1... so using mysql-router you always talk to 127.0.0.1.